core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceSettingServiceUnitTests.scala

```diff
@@ -548,4 +580,38 @@ class WorkspaceSettingServiceUnitTests extends AnyFlatSpec with MockitoTestUtils
       )
     )
   }
+
+  it should "require a non-negative retention duration for GcpBucketSoftDelete settings" in {
+    val negativeDurationSetting = GcpBucketSoftDeleteSetting(
+      GcpBucketSoftDeleteConfig(-1)
+    )
+
+    val service = workspaceSettingServiceConstructor()
+
+    val exception = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(service.setWorkspaceSettings(workspace.toWorkspaceName, List(negativeDurationSetting)), Duration.Inf)
+    }
+    exception.errorReport.statusCode shouldBe Some(StatusCodes.BadRequest)
+    exception.errorReport.message should include("Invalid settings requested.")
+    exception.errorReport.causes should contain theSameElementsAs List(
+      ErrorReport("Invalid GcpBucketSoftDelete configuration: retention duration must be from 0 to 90 days.")
+    )
+  }
+
+  it should "require a retention duration no more than 90 days for GcpBucketSoftDelete settings" in {
+    val longDurationSetting = GcpBucketSoftDeleteSetting(
+      GcpBucketSoftDeleteConfig(999_999_999)
+    )
+
+    val service = workspaceSettingServiceConstructor()
+
+    val exception = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(service.setWorkspaceSettings(workspace.toWorkspaceName, List(longDurationSetting)), Duration.Inf)
+    }
+    exception.errorReport.statusCode shouldBe Some(StatusCodes.BadRequest)
+    exception.errorReport.message should include("Invalid settings requested.")
+    exception.errorReport.causes should contain theSameElementsAs List(
```

Comment:
Food for thought: It's good to avoid matching on strings in tests when determining if a particular error was thrown since this leads to fragility down the line (i.e, what happens if someone fixes a typo in an error message?). It's usually better to craft a custom, semantically clear exception type that exposes the needed properties against which a unit test may make assertions.

Feel free to investigate that approach, or leave this as-is, I don't think it's a deal breaker here.